### PR TITLE
Enable dynamic detection of sound card

### PR DESCRIPTION
**Pull request by padmashree9110**
Signed-off-by: pmandri <padmashree.mandri@intel.com>

### CODE REVIEW

**Review by balajim001 — APPROVED** [review]
LGTM

File: primary/audio_hw.c [hunk 1]
```diff
@@ -1327,16 +1336,25 @@ static int adev_open_input_stream(struct audio_hw_device *dev,
...
-    params = pcm_params_get(PCM_CARD_DEFAULT, PCM_DEVICE, PCM_IN);
-
-    if(params != NULL) {
-        adev->cardc = PCM_CARD_DEFAULT;
-    } else {
-        adev->cardc = get_pcm_card("Dummy");
+    adev->cardc = get_pcm_card("PCH");
+    if (adev->cardc != -1)
         params = pcm_params_get(adev->cardc, PCM_DEVICE, PCM_IN);
+    else {
```
Comment (gkdeepa) [inline primary/audio_hw.c:1342]: Please add minimal code to include the logic in a single OR condition

File: primary/audio_hw.c [hunk 2]
```diff
@@ -1478,7 +1496,15 @@ static int adev_open(const hw_module_t* module, const char* name,
     adev->hw_device.dump = adev_dump;
     adev->hw_device.get_microphones = adev_get_microphones;
 
-    snprintf(mixer_path,PATH_MAX,"/vendor/etc/mixer_paths_%d.xml", card);
+    card = get_pcm_card("PCH");
```
Comment (gkdeepa) [inline primary/audio_hw.c:1499]: Please add minimal code to include the logic in a single OR condition

File: primary/audio_hw.c [hunk 3]
```diff
@@ -1064,13 +1064,22 @@ static int adev_open_output_stream(struct audio_hw_device *dev,
 
     int ret;
 
-    params = pcm_params_get(PCM_CARD_DEFAULT, PCM_DEVICE, PCM_OUT);
-
-    if(params != NULL) {
-       adev->card = PCM_CARD_DEFAULT;
-    } else {
-        adev->card = get_pcm_card("Dummy");
+    adev->card = get_pcm_card("PCH");
```
Comment (gkdeepa) [inline primary/audio_hw.c:1067]: Please add minimal code to include the logic in a single OR condition

**Review by gkdeepa — APPROVED** [review]
Please make sure you revisit the patch for having minimal code for the conditions added.

**Pull request merged**